src/Microsoft.Extensions.Primitives/StringTokenizer.cs

```diff
@@ -31,21 +43,21 @@ public StringTokenizer(string value, char[] separators) : this((StringSegment)va
         /// <param name="separators">The characters to tokenize by.</param>
         public StringTokenizer(StringSegment value, char[] separators)
         {
-            if (value == null)
+            if (!value.HasValue)
             {
-                throw new ArgumentNullException(nameof(value));
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.value);
             }
             
             if (separators == null)
             {
-                throw new ArgumentNullException(nameof(separators));
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.separators);
             }
 
             _value = value;
             _separators = separators;
         }
 
-        public Enumerator GetEnumerator() => new Enumerator(ref this);
+        public Enumerator GetEnumerator() => new Enumerator(in _value, _separators);
```

Comment:
Does `in` help codegen in this case? In theory ctor should be inlined and parameter copy eliminated.